File: backend/src/order/objects/Order.scala
package delivery.order.objects

import delivery.shared.objects.{MerchantId, OrderId, OrderStatus, RiderId, UserId, Voucher}

final case class Order(
    id: OrderId,
    customerId: UserId,
    customerName: String,
    customerPhone: String,
    merchantId: MerchantId,
    riderId: Option[RiderId],
    items: List[OrderItem],
    totalAmount: Double,
    deliveryAddress: String,
    status: OrderStatus,
    placedAt: String,
    originalAmount: Double = 0,
    discountAmount: Double = 0,
    payableAmount: Double = 0,
    usedVoucher: Option[Voucher] = None,
    pointsAwarded: Int = 0
)

File: frontend/src/objects/order/Order.ts
import type { MerchantId, OrderId, RiderId, UserId, OrderStatus, RefundStatus } from '@/objects/shared/ids'
import type { Promotion } from '@/objects/shared/Promotion'
import type { Voucher } from '@/objects/shared/Voucher'

import type { OrderItem } from './OrderItem'

export interface Order {
  id: OrderId
  customerId: UserId
  customerName: string
  customerPhone: string
  merchantId: MerchantId
  riderId?: RiderId
  items: OrderItem[]
  totalAmount: number
  deliveryAddress: string
  status: OrderStatus
  placedAt: string
  originalAmount: number
  discountAmount: number
  payableAmount: number
  usedVoucher?: Voucher
  merchantDiscountAmount?: number
  platformDiscountAmount?: number
  merchantReceivableAmount?: number
  appliedPromotions?: Promotion[]
  pointsAwarded: number
  refundStatus?: RefundStatus | null
  refundReason?: string | null
  refundImageUrl?: string | null
  refundRequestedAt?: string | null
  refundMerchantReason?: string | null
  refundMerchantReviewedAt?: string | null
  refundAdminReason?: string | null
  refundedAt?: string | null
  customerNoteText?: string | null
  customerNoteImageUrl?: string | null
}
